aten/src/ATen/native/Repeat.h

```diff
@@ -2,21 +2,32 @@
 
 #include <ATen/ATen.h>
 
-namespace at { namespace native {
+namespace at {
+namespace native {
 
 template <typename index_t, void compute(index_t*, int64_t*, index_t*, int64_t)>
-static inline Tensor repeat_interleave_common(const Tensor &repeats) {
-  TORCH_CHECK(repeats.dim() == 1, "repeat_interleave only accept 1D vector as repeat");
+static inline Tensor repeat_interleave_common(
+    const Tensor& repeats,
+    c10::optional<int64_t> output_size) {
+  TORCH_CHECK(
+      repeats.dim() == 1, "repeat_interleave only accept 1D vector as repeat");
   TORCH_CHECK(
       repeats.scalar_type() == at::kLong || repeats.scalar_type() == at::kInt,
       "repeats has to be Long or Int tensor");
-  TORCH_CHECK((repeats >= 0).all().item<uint8_t>(), "repeats can not be negative");
   if (repeats.size(0) == 0) {
     return at::empty_like(repeats, LEGACY_CONTIGUOUS_MEMORY_FORMAT);
   }
   Tensor repeats_ = repeats.contiguous();
   Tensor cumsum = repeats.cumsum(0);
-  int64_t total = cumsum[-1].item<int64_t>();
+  int64_t total;
+  if (output_size.has_value()) {
+    total = output_size.value();
+  } else {
+    total = cumsum[-1].item<int64_t>();
+    TORCH_CHECK(
```

Comment:
I was not too happy to move this check here. Other two alternatives I could think of was:
1. Do some kind of async check here?
2. Add another flag called "unsafe", if set to True, skip check.

Please let me know if you have guidance/preference on this.

Reply:
There's no way to do an async check here, but you can add asserts to the kernel checking that repeats are non-negative and that allocated result size matches the total_size argument